crates/bevy_math/src/curve/easing.rs

```diff
@@ -0,0 +1,314 @@
+//! Module containing different [`Easing`] curves to control the transition between two values and
+//! the [`EasingCurve`] struct to make use of them.
+
+use std::marker::PhantomData;
+
+use crate::{
+    ops::{self, FloatPow},
+    prelude::CubicSegment,
+    VectorSpace,
+};
+
+use super::{Curve, FunctionCurve, Interval};
+
+/// A trait for [`Curves`] that maps the [unit interval]. These kinds of curves are used to create
+/// new curves which create a transition between two values. Easing curves are most commonly known
+/// from [CSS animations] but are also widely used in other fields.
+///
+/// [unit interval]: `Interval::UNIT`
+/// [`Curves`]: `Curve`
+/// [CSS animations]: https://developer.mozilla.org/en-US/docs/Web/CSS/easing-function
+pub trait Easing: Curve<f32> {}
+impl<C: Curve<f32>> Easing for C {}
+
+/// A [`Curve`] that is defined by a `start` and an `end` point, together with an [`EasingCurve`]
+/// to interpolate between the values over the [unit interval].
+///
+/// [unit interval]: `Interval::UNIT`
+#[derive(Clone, Debug)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "bevy_reflect", derive(bevy_reflect::Reflect))]
+pub struct EasingCurve<T, E>
+where
+    T: VectorSpace,
+    E: Easing,
+{
+    start: T,
+    end: T,
+    easing: E,
+}
+
+impl<T, E> Curve<T> for EasingCurve<T, E>
+where
+    T: VectorSpace,
+    E: Easing,
+{
+    #[inline]
+    fn domain(&self) -> Interval {
+        Interval::UNIT
+    }
+
+    #[inline]
+    fn sample_unchecked(&self, t: f32) -> T {
+        let domain = self.easing.domain();
+        let t = domain.start().lerp(domain.end(), t);
```

Comment:
We allow the domain to be the real numbers, and I think this would make t NaN.

Reply:
True, thanks again!